### DISCUSSION TITLE
Add time to ball save based on event

### DISCUSSION THREAD

**Discussion by bernarma** [missionpinball/missionpinball#146] [Q&A] [answered]
Before crafting my own extend time on ball save I would like to know if it is currently done somewhere else or I have overlooked this functionality.

Use Case: When player hits a specific shot, add a specific amount of time to an enabled/running ball save.

Is this possible, out-of-the-box, with the ball_save device or alternative is there a semi out-of-the-box feature/pattern that has already been implemented that someone could direct me to?

**Comment by avanwinkle** [comment, marked as answer]
This is not functionality that's been requested before and does not exist on ball saves directly!

It would be reasonable to extend ball saves to include control events to add/jump the timer, and could even pause/subtract/restart as well. Essentially the code from the Timer device would need to be duplicated over into Ball Save and the configs extended.

Now that I write that out, it occurs to me that a simpler out-of-the-box option would be to literally use a Timer :) The timer's control events could start/add/jump/pause/restart et cetera. The timer's _started_ event would be the ball save's _enable_events_ and the timer's _complete_ event would be the ball save's _disable_events_.

**Reply by bernarma** [reply]
I just wanted to make sure before I get stuck into the solution that something more out of the box existed or didn't. I'll most likely use the Timer based approach as logically it makes sense. Thanks for the response.
